src/testdir/test_options.vim

```diff
@@ -360,13 +360,17 @@ func Test_set_completion()
   call assert_equal("\"set invtabstop=", @:)
 
   " Expand options for 'spellsuggest'
-  call feedkeys(":set spellsuggest=best,file:xyz\<Tab>\<C-B>\"\<CR>", 'xt')
-  call assert_equal("\"set spellsuggest=best,file:xyz", @:)
+  call feedkeys(":set spellsuggest=file:test_options.v\<Tab>\<C-B>\"\<CR>", 'xt')
+  call assert_equal("\"set spellsuggest=file:test_options.vim", @:)
+  call feedkeys(":set spellsuggest=best,file:test_options.v\<Tab>\<C-B>\"\<CR>", 'xt')
+  call assert_equal("\"set spellsuggest=best,file:test_options.vim", @:)
 
   " Expand value for 'key'
   set key=abcd
   call feedkeys(":set key=\<Tab>\<C-B>\"\<CR>", 'xt')
   call assert_equal('"set key=*****', @:)
+  call feedkeys(":set key-=\<Tab>\<C-B>\"\<CR>", 'xt')
+  call assert_equal('"set key-=*****', @:)
   set key=
```

Comment:
This seems to also need `if has('+key')`. 'key' option doesn't work if `FEAT_CRYPT` isn't enabled.

Reply:
I would remove the whole completion for `set key` (I mean what sense does it make to autocomplete `*****`) and I also don't want to accidentally leak the crypt-key

Reply:
I was actually going to suggest that in a separate PR! Since I was looking into cmdline completion, I think `key` needs some hardening.

I was going to propose removing completion for `set key`, and also remove `set-=` / `set+=` / `set^=`. Those have some security ramifications since they allow you to guess a substring of the key. We can just keep `set=`.

Reply:
thanks, yes makes sense!